frb_codegen/src/generator/rust/ty_optional.rs

```diff
@@ -34,19 +34,18 @@ impl TypeRustGeneratorTrait for TypeOptionalGenerator<'_> {
         })
     }
 
+    fn self_access(&self, obj: String) -> String {
+        format!("({})", obj)
+    }
+
     fn convert_to_dart(&self, obj: String) -> String {
         let inner = TypeRustGenerator::new(
             *self.ir.inner.clone(),
             self.context.ir_file,
             self.context.config,
         );
         let obj = match inner.wrapper_struct() {
-            Some(wrapper) => format!(
-                "{}.map(|v| {}({}))",
-                obj,
-                wrapper,
-                inner.self_access("v".to_owned())
-            ),
+            Some(wrapper) => format!("{}.map(|v| {}(v))", obj, wrapper,),
```

Comment:
As far as I understand, an architecture has been created so that the inner type of optional field is always `IrTypeBoxed`. (I couldn't quite understand why.) Here, too, if we use `inner.self_access("v".to_owned())`, since it will go to `self_access` in `TypeBoxedGenerator`, the dereferenced value `*v` was returned from there, which gave the error that the external struct could not be dereferenced.

For now, I removed the parameter `inner.self_access("v".to_owned())` and then I've added a bit of a hacky method here, as it only covers mirrored optional fields.

Reply:
> I couldn't quite understand why

Well, for simplicity? Because `Optional<Boxed<sth>>` can be translated to "nullable pointer", while `Optional<SthThatIsNotABox>` is harder to translate (maybe also forcefully translate to a nullable pointer?). But anyway, feel free to fix this problem!